stock_move_location/wizard/stock_move_location.py

```diff
@@ -0,0 +1,178 @@
+    def _onchange_locations(self):
+        self._clear_lines()
+
+    def _clear_lines(self):
+        origin = self.origin_location_id
+        destination = self.destination_location_id
+        # there is `invalidate_cache` call inside the unlink
+        # which will clear the wizard - not cool.
+        # we have to keep the values somehow
+        self.stock_move_locaiton_line_ids.unlink()
+        self.origin_location_id = origin
+        self.destination_location_id = destination
+
+    def _get_locations_domain(self):
+        return [('usage', '=', 'internal')]
+
+    def action_move_location(self):
+        inventory_obj = self.env["stock.inventory"]
+        collected_inventory = inventory_obj.create(
+            self._get_collected_inventory_values()
+        )
+        collected_inventory.action_start()
+        self.set_inventory_lines(collected_inventory)
+        collected_inventory.move_stock()
+        return self._get_inventory_action(collected_inventory.id)
+
+    def _get_collected_inventory_name(self):
+        sequence = self.env['ir.sequence'].next_by_code(
+            'stock.inventory.move') or '/'
+        res = "{sequence}:{location_from}:{location_to}".format(
+            sequence=sequence,
+            location_from=self.origin_location_id.display_name,
+            location_to=self.destination_location_id.display_name,
+        )
+        return res or "/"
+
+    def _get_collected_inventory_values(self):
+        return {
+            "name": self._get_collected_inventory_name(),
+            "location_id": self.origin_location_id.id,
+            "inventory_type": "move",
+            "destination_location_id": self.destination_location_id.id,
+            "filter": "partial",
+        }
+
+    def _get_inventory_action(self, inventory_id):
+        action = self.env.ref("stock.action_inventory_form").read()[0]
+        form_view = self.env.ref("stock.view_inventory_form").id
+        action.update({
+            "view_mode": "form",
+            "views": [(form_view, "form")],
+            "res_id": inventory_id,
+        })
+        return action
+
+    def _get_stock_move_location_lines_values(self):
+        quants_obj = self.env["stock.quant"]
+        product_obj = self.env['product.product']
+
+        locations = self.env['stock.location'].search([
+            ('id', 'child_of', [self.origin_location_id.id])
+        ])
+        company = self.env['res.company']._company_default_get(
+            'stock.inventory')
+        domain = [
+            ('location_id', 'in', locations.ids),
+            ('company_id', '=', company.id),
+        ]
+        grouped_quants_data = quants_obj.read_group(
+            domain=domain,
+            fields=['product_id', 'lot_id'],
+            groupby=['product_id', 'lot_id'],
+            lazy=False,
+        )
+
+        product_data = []
+        for group in grouped_quants_data:
+            quants = quants_obj.search(group.get("__domain"))
```

Comment:
seems that `quants` is needed only for the sum: can't you select it straight instead of using sum+mapped?